bin/helpers/capabilityHelper.js

```diff
@@ -241,6 +241,15 @@ const validate = (bsConfig, args) => {
       logger.warn(Constants.validationMessages.SPEC_TIMEOUT_NOT_PASSED_ERROR);
     }
 
+    if(!Utils.isUndefined(bsConfig.run_settings["record"]) && String(bsConfig.run_settings["record"]) == 'true') {
+      if(Utils.isUndefined(bsConfig.run_settings.projectId)) {
```

Comment:
let's add check for `projectId` or `record-key` being empty strings as well and show the warning.